cmd/root.go

```diff
@@ -47,104 +46,168 @@ var RootCmd = &cobra.Command{
 
-		go func() {
-			defer wg.Done()
-			image2, err = utils.ImagePrepper{img2Arg}.GetImage()
-			if err != nil {
-				glog.Error(err.Error())
-				os.Exit(1)
-			}
-		}()
+	var image1, image2 utils.Image
+	var err error
+	go func() {
+		defer wg.Done()
+		image1, err = utils.ImagePrepper{image1Arg}.GetImage()
+		if err != nil {
+			glog.Error(err.Error())
+			os.Exit(1)
+		}
+	}()
 
-		diffTypes, err := differs.GetDiffers(diffArgs)
+	go func() {
+		defer wg.Done()
+		image2, err = utils.ImagePrepper{image2Arg}.GetImage()
 		if err != nil {
 			glog.Error(err.Error())
 			os.Exit(1)
 		}
-		wg.Wait()
-
-		req := differs.DiffRequest{image1, image2, diffTypes}
-		if diffs, err := req.GetDiff(); err == nil {
-			// Outputs diff results in alphabetical order by differ name
-			diffTypes := []string{}
-			for name := range diffs {
-				diffTypes = append(diffTypes, name)
-			}
-			sort.Strings(diffTypes)
-			glog.Info("Retrieving diffs")
-			diffResults := []utils.DiffResult{}
-			for _, diffType := range diffTypes {
-				diff := diffs[diffType]
-				if json {
-					diffResults = append(diffResults, diff.GetStruct())
-				} else {
-					err = diff.OutputText(diffType)
-					if err != nil {
-						glog.Error(err)
-					}
-				}
-			}
+	}()
+
+	diffTypes, err := differs.GetAnalyzers(diffArgs)
+	if err != nil {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+	wg.Wait()
+
+	req := differs.DiffRequest{image1, image2, diffTypes}
+	if diffs, err := req.GetDiff(); err == nil {
+		// Outputs diff results in alphabetical order by differ name
+		sortedTypes := []string{}
+		for name := range diffs {
+			sortedTypes = append(sortedTypes, name)
+		}
+		sort.Strings(sortedTypes)
+		glog.Info("Retrieving diffs")
+		diffResults := []utils.DiffResult{}
+		for _, diffType := range sortedTypes {
+			diff := diffs[diffType]
 			if json {
-				err = utils.JSONify(diffResults)
+				diffResults = append(diffResults, diff.GetStruct())
+			} else {
+				err = diff.OutputText(diffType)
 				if err != nil {
 					glog.Error(err)
 				}
 			}
-			fmt.Println()
+		}
+		if json {
+			err = utils.JSONify(diffResults)
+			if err != nil {
+				glog.Error(err)
+			}
+		}
+		fmt.Println()
+		if !save {
 			glog.Info("Removing image file system directories from system")
-			if !save {
-				errMsg := remove(image1.FSPath, true)
-				errMsg += remove(image2.FSPath, true)
-				if errMsg != "" {
-					glog.Error(errMsg)
-				}
+			errMsg := remove(image1.FSPath, true)
+			errMsg += remove(image2.FSPath, true)
+			if errMsg != "" {
+				glog.Error(errMsg)
+			}
+		} else {
+			dir, _ := os.Getwd()
+			glog.Infof("Images were saved at %s as %s and %s", dir, image1.FSPath, image2.FSPath)
+		}
+	} else {
+		glog.Error(err.Error())
+		os.Exit(1)
+	}
+}
+
+func analyzeImage(imageArg string, analyzerArgs []string) {
+	image, err := utils.ImagePrepper{imageArg}.GetImage()
+	if err != nil {
+		glog.Error(err.Error())
+		os.Exit(1)
```

Comment:
not sure if this is new to this PR, but `GetImage()` extracts the image filesystem and saves it on disk right? if so, calling `os.Exit(1)` here will skip the cleanup of that image if the user didn't want it around. probably want to defer an image cleanup in the root command, and then return an error here.